src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -16,7 +17,7 @@ public interface MachineService {
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<T> getAll(Class<? extends Machine> type);
```

Comment:
Parameterize `MachineService` to ensure that the `getAll` method can only accept a `Class` type that is a subtype of `T`. This will enforce type safety and prevent passing unrelated class types to the method.